base/gmp.jl

```diff
@@ -69,6 +69,93 @@ function __init__()
     end
 end
 
+
+module mpz
+# wrapping of libgmp functions
+# "output parameters" are labeled x, y, z, and are returned when appropriate
+# constant input parameter are labeled a, b, c
+
+using Base.GMP: BigInt, Limb
+
+gmpz(op::Symbol) = (Symbol(:__gmpz_, op), :libgmp)
+
+sizeinbase(a::BigInt, b) = Int(ccall((:__gmpz_sizeinbase, :libgmp), Csize_t, (Ptr{BigInt}, Cint), &a, b))
+
+for op in (:add, :sub, :mul, :fdiv_q, :tdiv_q, :fdiv_r, :tdiv_r, :gcd, :lcm, :and, :ior, :xor)
+    @eval begin
+        $op(x::BigInt, a::BigInt, b::BigInt) = (ccall($(gmpz(op)), Void, (Ptr{BigInt}, Ptr{BigInt}, Ptr{BigInt}), &x, &a, &b); x)
+        $op(a::BigInt, b::BigInt) = $op(BigInt(), a, b)
+        $(Symbol(op, :!))(x::BigInt, a::BigInt) = $op(x, x, a)
+    end
+end
+
+invert(x::BigInt, a::BigInt, b::BigInt) =
+    ccall((:__gmpz_invert, :libgmp), Cint, (Ptr{BigInt}, Ptr{BigInt}, Ptr{BigInt}), &x, &a, &b)
```

Comment:
If this is getting updated might as well update the   `&` syntax , since it is deprecated and should be using  the `Ref{T}` argument type instead.